src/Microsoft.Identity.Client/PublicClientApplication.cs

```diff
@@ -283,15 +283,17 @@ public async Task<AuthenticationResult> AcquireTokenAsync(IEnumerable<string> sc
 
         private static object GetParentObjectFromUiParent(UIParent parent)
         {
-            #if DESKTOP
-                return parent.CoreUIParent.OwnerWindow;
-            #elif ANDROID
-                return parent.CoreUIParent.Activity;
-            #elif MAC
-                return parent.CoreUIParent.CallerWindow;
-            #else
-                return null;
-            #endif
+#if DESKTOP
+            return parent.CoreUIParent.OwnerWindow;
+#elif ANDROID
+            return parent.CoreUIParent.Activity;
+#elif MAC
+            return parent.CoreUIParent.CallerWindow;
+#elif iOS
+            return parent.CoreUIParent.CallerViewController;
```

Comment:
we need the ViewController to see if we can open broker. I need to see if this will impact our non-broker scenarios (esp. the ones using system browser).